src/providers/export/azureCustomVision/azureCustomVisionService.ts

```diff
@@ -0,0 +1,168 @@
+     * @param project - The Azure Custom Vision Project settings
+     */
+    public async create(project: IAzureCustomVisionProject): Promise<IAzureCustomVisionProject> {
+        Guard.null(project);
+
+        const urlParams = {
+            name: project.name,
+            description: project.description,
+            projectType: project.projectType,
+            domainId: project.domainId,
+            classificationType: project.classificationType,
+        };
+
+        const url = `${this.options.baseUrl}/projects?${createQueryString(urlParams)}`;
+        const response = await axios.post(url, null, this.createRequestConfig());
+
+        if (response.status !== 200) {
+            throw new Error("Error creating new project");
+        }
+
+        return response.data as IAzureCustomVisionProject;
+    }
+
+    /**
+     * Get the tags of a Azure Custom Vision project
+     * @param projectId The Azure Custom Vision project ID
+     */
+    public async getProjectTags(projectId: string): Promise<IAzureCustomVisionTag[]> {
+        Guard.emtpy(projectId);
+
+        const url = `${this.options.baseUrl}/projects/${projectId}/tags`;
+        const response = await axios.get(url, this.createRequestConfig());
+
+        if (response.status !== 200) {
+            throw new Error("Error retrieving project tags");
+        }
+
+        return response.data as IAzureCustomVisionTag[];
+    }
+
+    /**
+     * Creates a new Azure Custom Vision Tag
+     * @param projectId The Azure Custom vision project id
+     * @param tag The tag to save
+     */
+    public async createTag(projectId: string, tag: IAzureCustomVisionTag): Promise<IAzureCustomVisionTag> {
+        Guard.emtpy(projectId);
+        Guard.null(tag);
+
+        const urlParams = {
+            name: tag.name,
+            description: tag.description,
+        };
+
+        const url = `${this.options.baseUrl}/projects/${projectId}/tags?${createQueryString(urlParams)}`;
+        const response = await axios.post(url, null, this.createRequestConfig());
+
+        if (response.status !== 200) {
+            throw new Error("Error saving tag");
+        }
+
+        return response.data as IAzureCustomVisionTag;
+    }
+
+    /**
+     * Uploads a new image to an Azure Custom Vision project
+     * @param projectId The Azure Custom vision project id
+     * @param contents The asset binary contents
+     */
+    public async createImage(projectId: string, contents: File | Blob | ArrayBuffer): Promise<IAzureCustomVisionImage> {
+        Guard.emtpy(projectId);
+        Guard.null(contents);
+
+        const url = `${this.options.baseUrl}/projects/${projectId}/images`;
+        const config = this.createRequestConfig();
+        config.headers["Content-Type"] = "application/octet-stream";
+        const response = await axios.post(url, contents, config);
+
+        if (response.status !== 200) {
+            throw new Error("Error saving image");
```

Comment:
More of a question - do we add localization for error messages?

Reply:
Will run a localization pass with a new PR to catch all of these.